test/cluster/mzcompose.py

```diff
@@ -599,186 +597,6 @@ def workflow_test_resource_limits(c: Composition) -> None:
         c.run("testdrive", "resources/resource-limits.td")
 
 
-def workflow_test_builtin_migration(c: Composition) -> None:
```

Comment:
Sad to see this go because it caught a lot of bugs in the past, but I guess the idea is that the upgrade tests should serve the same purpose?

Reply:
cc @philip-stoev

Reply:
I'm sad to lose it too. I filed https://github.com/MaterializeInc/materialize/issues/17236 about bringing it back, so we don't forget. Going to merge for now though to make progress on cluster unification.

Reply:
Yes, the existing upgrade tests should trigger migrations. Do you recall a regression that caused this test to fail while leaving the upgrade tests passing? I would be very much interested to debug why that happened.

Reply:
They only trigger built in table migrations if the current_source has a builtin table with a new column relative to the last release. That means we only get coverage of the builtin table migration code on the weeks where we have such a change; and the code could regress on the weeks where we don't.

Reply:
@benesch ouch ... if we never test upgrade from a release that is more than 2 approx weeks old, how do we close that hole in testing? At the same time, the current upgrade tests will test the same upgrade path that can happen on the cloud, where there will also never be releases older than 2 weeks, so there is no hole in testing?

Reply:
> Yes, the existing upgrade tests should trigger migrations. Do you recall a regression that caused this test to fail while leaving the upgrade tests passing? I would be very much interested to debug why that happened.

I do not recall whether or not this ever happened. This is a bit tricky, because if we're committing to only ever upgrading to the next immediate major version, then it feels a bit silly to test an upgrade that skips a bunch of versions. However, on the other hand, if we have a release that changes the builtin migration code AND doesn't change the schema of any builtin object then there is no way to test the code changes. 

Maybe we want to have a test like the old `test_builtin_migrations`, however it performs multiple upgrades. One upgrade for every version in-between the version where the schema change happened to the current version. That test would likely be very slow but would be able to test builtin migrations and conform to the practice of only doing upgrades to the next immediate major version.